### ISSUE TITLE
display() - stash to cloudant fails: Database dataframe-20170227-153015 does not exist or is not accessible

### ISSUE THREAD

**Issue by ptitzler** [ibm-cds-labs/pixiedust#174]
PD 0.84: (configured using David's cloudant-spark.jar)
```
import pixiedust
df=pixidiest.sampleData(6)
display(df)
```

* Select Stash to Cloudant 
* Create a valid connection profile for a Cloudant instance of your choice. The instance should not contain any databases starting with `dataframe`
* Select the profile
* Submit

An error is raised:
```
---------------------------------------------------------------------------\nPy4JJavaError                             Traceback (most recent call last)\n<ipython-input-4-64feb3e4cf88> in <module>()
----> 1 display(df,cell_id='0669EA96AEFB438989BAC974E7CBCBBA',showLegend='false',mpld3='false',aggregation='SUM',logx='false',clusterby='CITY',rowCount='1',handlerId='stashCloudant',valueFields='PRICE',rendererId='matplotlib',keyFields='STATE',nostore_cw='1098',nostore_bokeh='false',prefix='196a07e2',nostore_dbName='dataframe-20170227-153015',nostore_connection='test1')
\n/Users/patti/pixiedust_dev/pixiedust/pixiedust/display/__init__.pyc in display(entity, **kwargs)
    128                 return
    129 
--> 130             displayHandler.render()
    131         finally:
    132             if pr is not None:
\n/Users/patti/pixiedust_dev/pixiedust/pixiedust/display/display.pyc in render(self)
    223         else:
    224             start = time.clock()
--> 225             self.doRender(handlerId)
    226             self.executionTime = time.clock() - start
    227 
\n/Users/patti/pixiedust_dev/pixiedust/pixiedust/services/stashCloudant.pyc in doRender(self, handlerId)
     64                     .option("cloudant.password",credentials["password"])\
     65                     .option("createDBOnSave","true")\
---> 66                     .save(dbName)
     67                 print("""Successfully stashed your data: <a target='_blank' href='{0}/{1}'>{1}</a>""".format(credentials["url"],dbName))
\n/Users/patti/pixiedust_dev/spark1.6.0/python/pyspark/sql/readwriter.pyc in save(self, path, format, mode, partitionBy, **options)
    385             self._jwrite.save()
    386         else:
--> 387             self._jwrite.save(path)
    388 
    389     @since(1.4)
\n/Users/patti/pixiedust_dev/spark1.6.0/python/lib/py4j-0.9-src.zip/py4j/java_gateway.py in __call__(self, *args)
    811         answer = self.gateway_client.send_command(command)
    812         return_value = get_return_value(
--> 813             answer, self.gateway_client, self.target_id, self.name)
    814 
    815         for temp_arg in temp_args:
\n/Users/patti/pixiedust_dev/spark1.6.0/python/pyspark/sql/utils.pyc in deco(*a, **kw)
     43     def deco(*a, **kw):
     44         try:
---> 45             return f(*a, **kw)
     46         except py4j.protocol.Py4JJavaError as e:
     47             s = e.java_exception.toString()
\n/Users/patti/pixiedust_dev/spark1.6.0/python/lib/py4j-0.9-src.zip/py4j/protocol.py in get_return_value(answer, gateway_client, target_id, name)
    306                 raise Py4JJavaError(
    307                     "An error occurred while calling {0}{1}{2}.\n".
--> 308                     format(target_id, ".", name), value)
    309             else:
    310                 raise Py4JError(
\nPy4JJavaError: An error occurred while calling o51.save.
: java.lang.RuntimeException: Database dataframe-20170227-153015 does not exist or is not accessible. Failed to save data!
	at com.cloudant.spark.common.JsonStoreDataAccess$$anonfun$saveAll$1.apply$mcVI$sp(JsonStoreDataAccess.scala:249)
	at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:141)
	at com.cloudant.spark.common.JsonStoreDataAccess.saveAll(JsonStoreDataAccess.scala:214)
	at com.cloudant.spark.CloudantReadWriteRelation.insert(DefaultSource.scala:65)
	at com.cloudant.spark.DefaultSource.createRelation(DefaultSource.scala:100)
	at com.cloudant.spark.DefaultSource.createRelation(DefaultSource.scala:72)
	at org.apache.spark.sql.execution.datasources.ResolvedDataSource$.apply(ResolvedDataSource.scala:222)
	at org.apache.spark.sql.DataFrameWriter.save(DataFrameWriter.scala:148)
	at org.apache.spark.sql.DataFrameWriter.save(DataFrameWriter.scala:139)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:231)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:381)
	at py4j.Gateway.invoke(Gateway.java:259)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:133)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:209)
	at java.lang.Thread.run(Thread.java:745)
\n
```

**Note that a database named `dataframe-20170227-153015` was created in the selected Cloudant service instance and NNN records were saved.** (NNN varies)

**Comment by DTAIEB** [comment]
I can't seem to reproduce this issue. perhaps it's been fixed with #173

**Comment by DTAIEB** [comment]
Proper fix in #177

**Comment by ptitzler** [comment]
:+1: No longer reproducible.
